### DISCUSSION TITLE
phase molecular weight derivative calculations

### DISCUSSION THREAD

**Discussion by ver2king** [GEOS-DEV/GEOS#2318] [Q&A] [answered]
My concern is within the following file, and the code snippets below:

https://github.com/GEOSX/GEOSX/blob/cadc47703f3936f8cff304a10c7ece9168b4c3cc/src/coreComponents/constitutive/fluid/CO2BrineFluid.hpp

    // 4.1 Compute the phase molecular weights (ultimately, get that from the PVT function)

    real64 phaseMolecularWeight[numPhase]{};
    real64 dPhaseMolecularWeight[numPhase][numComp+2]{};

    real64 phaseMolarDens{};
    stackArray1d< real64, numComp+2 > dPhaseMolarDens( numComp+2 );

    m_phase1.density.compute( pressure,
                              temperatureInCelsius,
                              phaseCompFraction.value[ip1].toSliceConst(), phaseCompFraction.derivs[ip1].toSliceConst(),
                              phaseMolarDens, dPhaseMolarDens.toSlice(),
                              false );
    phaseMolecularWeight[ip1] = phaseDensity.value[ip1] / phaseMolarDens;
    for( integer idof = 0; idof < numComp+2; ++idof )
    {
      dPhaseMolecularWeight[ip1][idof] = phaseDensity.derivs[ip1][idof] / phaseMolarDens - phaseMolecularWeight[ip1] * dPhaseMolarDens[idof] / phaseMolarDens;
    }

    m_phase2.density.compute( pressure,
                              temperatureInCelsius,
                              phaseCompFraction.value[ip2].toSliceConst(), phaseCompFraction.derivs[ip2].toSliceConst(),
                              phaseMolarDens, dPhaseMolarDens.toSlice(),
                              false );
    phaseMolecularWeight[ip2] = phaseDensity.value[ip2] / phaseMolarDens;
    for( integer idof = 0; idof < numComp+2; ++idof )
    {
      dPhaseMolecularWeight[ip2][idof] = phaseDensity.derivs[ip2][idof] / phaseMolarDens - phaseMolecularWeight[ip2] * dPhaseMolarDens[idof] / phaseMolarDens;
    }

I could not interpret the formula behind these derivative calculations nor locate anywhere in the GEOSX documetation that mentions this aspect. May anyone help me understand the formulations behind those calculations?

**Comment by francoishamon** [comment]
What about discussing all your questions at once, on Zoom? That will be more efficient
Would that work for you?

**Comment by ver2king** [comment, marked as answer]
@francoishamon that shall work for me, and I do appreciate your swift response. Please just inform me of your availability and I shall share the Zoom meeting link.

**Reply by francoishamon** [reply]
Monday, Tuesday, or Wednesday afternoon next week?

**Reply by ver2king** [reply]
Just as a side note, I am located in the US CST time zone. Do you mean US CST or another time zone?

**Reply by francoishamon** [reply]
I meant US Pacific Time if that works for you

**Reply by ver2king** [reply]
Then, nevermind, not a huge difference to be fair. I am available on Monday & Wednesday afternoons next week. Therefore, I shall send the Zoom link for Monday afternoon sometime tomorrow (i.e., Friday 03/03/2023).
